r/inst/build_arrow_static.sh

```diff
@@ -55,6 +55,7 @@ ${CMAKE} -DARROW_BOOST_USE_SHARED=OFF \
     -DARROW_BUILD_TESTS=OFF \
     -DARROW_BUILD_SHARED=OFF \
     -DARROW_BUILD_STATIC=ON \
+    -DARROW_ACERO=${ARROW_ACERO:-$ARROW_DEFAULT_PARAM} \
```

Comment:
How about using `ON` as the default value because `ARROW_DATASET` uses `ON` as the default value?

```suggestion
    -DARROW_ACERO=${ARROW_ACERO:-ON} \
```

Reply:
That is the main question in https://github.com/apache/arrow/issues/34887

`ARROW_DEFAULT_PARAM=ON` unless `LIBARROW_MINIMAL` is set.  If I change this to `ON` then there will be no testing where this is off.

However, your point is valid.  It would not make sense for `ARROW_DATASET` to be `ON` and `ARROW_ACERO` to be `OFF`.  I will change this back to `ON` and we can fix the testing as part of #34887 if we want to support this.